recipes/mysql/build.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+# this script is based off the homebrew package:
+# https://github.com/Homebrew/homebrew-core/blob/master/Formula/mysql.rb
+
+# Seems to be needed to build in some spots.
+for pth in /lib/cpp /lib64/cpp /usr/bin/cpp
+do
+    if [ ! -e ${pth} ]
+    then 
+        mkdir -p `dirname ${pth}`
+        ln -s `which cpp` ${pth}
+    fi
+done
+
+mkdir -p build
+cd build
+
+# -DINSTALL_* are relatiove to -DCMAKE_INSTALL_PREFIX
+mkdir -p ${PREFIX}/mysql
+cmake \
+    -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+    -DMYSQL_DATADIR=${PREFIX}/mysql/datadir \
+    -DINSTALL_INCLUDEDIR=include/mysql \
+    -DINSTALL_MANDIR=share/man \
+    -DINSTALL_DOCDIR=share/doc/mysql \
+    -DINSTALL_DOCREADMEDIR=mysql \
+    -DINSTALL_INFODIR=share/info \
+    -DINSTALL_MYSQLSHAREDIR=share/mysql \
+    -DINSTALL_SUPPORTFILESDIR=mysql/support-files \
+    -DINSTALL_SCRIPTDIR=mysql/scripts \
+    -DSYSCONFDIR=${PREFIX}/etc \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_FIND_FRAMEWORK=LAST \
+    -DCMAKE_VERBOSE_MAKEFILE=ON \
+    -Wno-dev \
+    -DWITH_UNIT_TESTS=OFF \
+    -DDEFAULT_CHARSET=utf8 \
+    -DDEFAULT_COLLATION=utf8_general_ci \
+    -DCOMPILATION_COMMENT=conda-forge \
+    -DWITH_SSL=bundled \
+    -DWITH_EDITLINE=bundled \
+    -DWITH_READLINE=bundled \
+    -DWITH_BOOST=boost \
+    -DDOWNLOAD_BOOST=1 \
+    ..
+
+make
+make install
+
+# we will run this test now and then delete the directory
+# there is no reason to ship the test dir and it is big
```

Comment:
You could use [`test/source_files`](https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#source-files) to let this happen in the test step. But testing here is probably fine too, especially since you're still testing that the commands run okay.